### PR TITLE
Bdd tests

### PR DESCRIPTION
**Pull request by gemmai95**
Implement scenarios added by Mark

### CODE REVIEW

File: tests/behavioural/features/steps/draft_post.py [hunk 1]
```diff
@@ -154,6 +156,35 @@ def step_impl(context):
...
+
+@when("the user navigates to the draft inbox")
+def step_impl(context):
+    request = app.test_client().get("http://localhost:5050/draft/{0}".format(context.msg_id))
+    context.request_data = json.loads(request.data)
+
+
+@then("the draft message is displayed in the draft inbox")
+def step_impl(context):
+    nose.tools.assert_true(context.request_data is not None)
```
Comment (AndrewTorrance) [inline tests/behavioural/features/steps/draft_post.py]: Shouldnt this be testing response ? And also shouldnt it test more than just not None ?

File: tests/behavioural/features/steps/message_put.py [hunk 2]
```diff
@@ -247,140 +247,143 @@ def step_impl(context):
...
-    response = json.loads(context.response.data)
-    msg_id = response['msg_id']
     token_data['user_urn'] = data['urn_from']
     headers['authentication'] = update_encrypted_jwt()
-    response_get = app.test_client().get("http://localhost:5050/message/{0}".format(msg_id), headers=headers)
+    response_get = app.test_client().get("http://localhost:5050/message/{0}".format(context.msg_id), headers=headers)
     context.get_json = json.loads(response_get.data)
 
 
-@then("the status of the message changes to from 'unread' to 'read' for all internal users that have access to that survey")
```
Comment (AndrewTorrance) [inline tests/behavioural/features/steps/message_put.py]: This tests for none being unread , not all being read

File: tests/behavioural/features/steps/message_put.py [hunk 3]
```diff
@@ -247,140 +247,143 @@ def step_impl(context):
...
-@then("the status of that message changes to 'read'")
-def step_impl(context,step):
-    response_get = app.test_client().get("http://localhost:5050/message/{0}".format(context.msg_id), headers=headers)
-    data_get = json.loads(response_get.data)
-    nose.tools.assert_true("ARCHIVE" not in data_get['labels'])
-    return step.error_message
+@then("the status of the message changes to from unread to read")
+def step_impl(context):
+    nose.tools.assert_true("UNREAD" not in context.request_data['labels'])
+
```
Comment (AndrewTorrance) [inline tests/behavioural/features/steps/message_put.py]: This is testing NOT UNREAD , it does not check READ

File: tests/behavioural/features/steps/message_put.py [hunk 4]
```diff
@@ -247,140 +247,143 @@ def step_impl(context):
...
+    modify_data['action'] = 'remove'
+    modify_data['label'] = 'UNREAD'
+    app.test_client().put(url.format(context.msg_id), data=flask.json.dumps(modify_data), headers=headers)
 
 
+@then("the status of that message changes to read")
+def step_impl(context):
+    request = app.test_client().get("http://localhost:5050/message/{0}".format(context.msg_id), headers=headers)
+    request_data = json.loads(request.data)
+    nose.tools.assert_true("UNREAD" not in request_data['labels'])
```
Comment (AndrewTorrance) [inline tests/behavioural/features/steps/message_put.py]: Tests for not unread , not read
